src/test/kotlin/org/rust/toml/CargoTomlDependenciesCompletionTest.kt

```diff
@@ -0,0 +1,164 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.toml
+
+import org.intellij.lang.annotations.Language
+
+class CargoTomlDependenciesCompletionTest : CargoTomlCompletionTestBase() {
```

Comment:
Now dependency completion works with complex keys like `target.'cfg(windows)'.dependencies`  and `target.'cfg(windows)'.dependencies.dep`.
But we still don't have the corresponding tests. Let's add them.

Reply:
These cases are covered by `CargoTomlPsiPatternTest`. Should be double-check it?

Reply:
But it's not really true.
Yeah, we indeed have tests for patterns. But are we sure that we correctly use these patterns in other code to make completion works in a necessary way?
In my opinion, pattern tests are convenient way for developers to find out a problem when tests fail. User know nothing about patterns but he wants that completion works. So, first of all, we should test the user use case and other tests are optional/auxiliary

Reply:
Ok, I've added this test:
```kotlin
fun `test empty key with complex dependency head`() = doTest("""
        [target.'cfg(windows)'.dev-dependencies]
        <caret>
    """, """
        [target.'cfg(windows)'.dev-dependencies]
        dep = "1.0"
    """, "dep" to "1.0")
```

Reply:
👍